cargo-lock/tests/lockfile.rs

```diff
@@ -198,155 +198,43 @@ mod tree {
     }
 }
 
-/// Test that a lockfile with ambiguous registries have registries encoded for each package which has multiple
 #[test]
-fn encoding_multi_registry() {
-    let lockfile = cargo_lock::Lockfile::from_str(
-        r#"# This file is automatically @generated by Cargo.
-# It is not intended for manual editing.
-version = 3 
-
-[[package]]
-name = "bytes"
```

Comment:
Sorry if I wasn't that clear before: I think all existing test cases should be moved to the new mechanism in the first commit.

Reply:
Hmm, the problem is that the `encoding_multi_registry()` doesn't really fit within the new mechanism. That test takes a lock file _without_ source specs for unambiguous dependency specs, passes it through `cargo-lock` and then expects a lock file _with_ source specs for unambiguous dependency specs. In the new mechanism, the input to and the output from `cargo-lock` need to be the same.

If you want, I can move the input of `encoding_multi_registry()` to a separate file and change the test, but then the commit is not a pure refactoring, it already changes the test. Would you prefer that?

Reply:
> If you want, I can move the input of `encoding_multi_registry()` to a separate file and change the test, but then the commit is not a pure refactoring, it already changes the test. Would you prefer that?

Yes, I think that would work. I'd like to be able to review in isolation how the old test is being represented in the new situation.